packages/agreement-process/src/services/agreementStateProcessor.ts

```diff
@@ -144,3 +169,179 @@ export const suspendedByProducerFlag = (
+  consumer: CompactTenant,
+  eservices: EService[],
+  correlationId: string,
+  logger: Logger
+): Promise<CreateEvent<AgreementEvent> | void> {
+  const descriptor = eservices
+    .find((eservice) => eservice.id === agreement.data.eserviceId)
+    ?.descriptors.find(
+      (descriptor) => descriptor.id === agreement.data.descriptorId
+    );
+
+  if (!descriptor) {
+    logger.error(
+      `Descriptor ${agreement.data.descriptorId} not found for Agreement ${agreement.data.id} - EService ${agreement.data.eserviceId}`
+    );
+    return;
+  }
+
+  const nextState = nextStateByAttributes(agreement.data, descriptor, consumer);
+
+  const newSuspendedByPlatform = suspendedByPlatformFlag(nextState);
+
+  const finalState = agreementStateByFlags(
+    nextState,
+    agreement.data.suspendedByProducer,
+    agreement.data.suspendedByConsumer,
+    newSuspendedByPlatform
+  );
+
+  const updatedAgreement: Agreement = {
+    ...agreement.data,
+    state: finalState,
+    suspendedByPlatform: newSuspendedByPlatform,
+  };
+
+  if (
+    allowedStateTransitions(agreement.data.state).includes(finalState) &&
+    newSuspendedByPlatform !== agreement.data.suspendedByPlatform
+  ) {
+    return match(finalState)
+      .with(agreementState.suspended, () =>
+        toCreateEventAgreementSuspendedByPlatform(
+          updatedAgreement,
+          agreement.metadata.version,
+          correlationId
+        )
+      )
+      .with(agreementState.active, () =>
+        toCreateEventAgreementUnsuspendedByPlatform(
+          updatedAgreement,
+          agreement.metadata.version,
+          correlationId
+        )
+      )
+      .with(agreementState.missingCertifiedAttributes, () =>
+        toCreateEventAgreementPutInMissingCertifiedAttributesByPlatform(
+          updatedAgreement,
+          agreement.metadata.version,
+          correlationId
+        )
+      )
+      .with(agreementState.draft, () =>
+        toCreateEventAgreementPutInDraftByPlatform(
+          updatedAgreement,
+          agreement.metadata.version,
+          correlationId
+        )
+      )
+      .otherwise(
+        () =>
+          void logger.error(
+            `Agreement state transition not allowed from ${agreement.data.state} to ${finalState}`
+          )
+      );
+  } else {
+    logger.error(
+      `Agreement state transition not allowed from ${agreement.data.state} to ${finalState}`
+    );
+  }
+}
+
+function eserviceContainsAttribute(
+  attributeId: AttributeId,
+  eservice: EService
+): boolean {
+  const certified = eservice.descriptors.flatMap(
+    (descriptor) => descriptor.attributes.certified
+  );
+  const declared = eservice.descriptors.flatMap(
+    (descriptor) => descriptor.attributes.declared
+  );
+  const verified = eservice.descriptors.flatMap(
+    (descriptor) => descriptor.attributes.verified
+  );
+
+  const allIds = [...certified, ...declared, ...verified].flatMap((attr) =>
+    attr.map((a) => a.id)
+  );
```

Comment:
```suggestion
  const allIds = eservice.descriptors
    .flatMap((descriptor) => [
      ...descriptor.attributes.certified,
      ...descriptor.attributes.declared,
      ...descriptor.attributes.verified,
    ])
    .flatMap((attr) => attr.map((a) => a.id));
```
We can simplify it like this